### PR TITLE
[clang-analyze] ServiceBroker EventLog use pointer instead of reference

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Clang-analyzer detects the possibility of potential object pointer being null in
CServiceBroker::GetEventLog(). As we return a reference, its expected a reference is never
null. change to a pointer and do null checks

```
macos/xbmc/ServiceBroker.cpp:253:10: warning: Called C++ object pointer is null
  return g_serviceBroker.m_pSettingsComponent->GetProfileManager()->GetEventLog();
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

## Motivation and context
Clang-analyze warning fixes.
Not sure if this would be the best way, but the m_pSettingsComponent can potentially be nullptr (std::unique_ptr) as well as GetProfileManager (std::shared_ptr).

## How has this been tested?
build/runtime osx. Eventlog still logs/deletes entries

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by DaveTBlake — APPROVED** [review]

File: xbmc/ServiceBroker.cpp [hunk 1]
```diff
@@ -248,9 +248,16 @@ CDatabaseManager& CServiceBroker::GetDatabaseManager()
...
+CEventLog* CServiceBroker::GetEventLog()
 {
-  return g_serviceBroker.m_pSettingsComponent->GetProfileManager()->GetEventLog();
+  if (g_serviceBroker.m_pSettingsComponent)
+  {
+    auto profileManager = g_serviceBroker.m_pSettingsComponent->GetProfileManager();
+    if (profileManager)
+      return &profileManager->GetEventLog();
+  }
+
```
Comment (howie-f) [inline xbmc/ServiceBroker.cpp:259]: should we log an error here? just in case

File: xbmc/ServiceBroker.cpp [hunk 2]
```diff
@@ -248,9 +248,16 @@ CDatabaseManager& CServiceBroker::GetDatabaseManager()
   return g_application.m_ServiceManager->GetDatabaseManager();
 }
 
-CEventLog& CServiceBroker::GetEventLog()
+CEventLog* CServiceBroker::GetEventLog()
 {
-  return g_serviceBroker.m_pSettingsComponent->GetProfileManager()->GetEventLog();
+  if (g_serviceBroker.m_pSettingsComponent)
```
Comment (lrusak) [inline xbmc/ServiceBroker.cpp:253]: Maybe use negative check here to reduce the nested if statements.

File: xbmc/addons/RepositoryUpdater.cpp [hunk 3]
```diff
@@ -94,8 +94,9 @@ void CRepositoryUpdater::OnJobComplete(unsigned int jobID, bool success, CJob* j
               "", g_localizeStrings.Get(24001), g_localizeStrings.Get(24061),
               TOAST_DISPLAY_TIME, false, TOAST_DISPLAY_TIME);
 
+        auto eventLog = CServiceBroker::GetEventLog();
         for (const auto &addon : updates)
-          CServiceBroker::GetEventLog().Add(EventPtr(new CAddonManagementEvent(addon, 24068)));
+          eventLog->Add(EventPtr(new CAddonManagementEvent(addon, 24068)));
```
Comment (lrusak) [inline xbmc/addons/RepositoryUpdater.cpp:99]: You don't check for nullptr before usage here.

**Review by lrusak — COMMENTED** [review]
Looks pretty good.

Reading these changes makes me think that someone should make EventPtr use a unique_ptr 🙃

File: xbmc/ServiceBroker.cpp (hunk 2 above)
Comment (fuzzard) [reply xbmc/ServiceBroker.cpp:253]: id end up having to do a return nullptr, and then still have the fallthrough return nullptr.
Is there any benefit?

```
CEventLog* CServiceBroker::GetEventLog()
{
  if (!g_serviceBroker.m_pSettingsComponent)
    return nullptr;
  
  auto profileManager = g_serviceBroker.m_pSettingsComponent->GetProfileManager();
  if (profileManager)
    return &profileManager->GetEventLog();
  
  return nullptr;
}
```

Comment (lrusak) [reply xbmc/ServiceBroker.cpp:253]: What about this?

```
CEventLog* CServiceBroker::GetEventLog()
{
  if (!g_serviceBroker.m_pSettingsComponent)
    return nullptr;
  
  auto profileManager = g_serviceBroker.m_pSettingsComponent->GetProfileManager();
  if (!profileManager)
    return nullptr;

  return &profileManager->GetEventLog();
}

```

Comment (fuzzard) [reply xbmc/ServiceBroker.cpp:253]: Sure, im easy.

**Comment by fuzzard** [comment]
> Looks pretty good.
> 
> Reading these changes makes me think that someone should make EventPtr use a unique_ptr 🙃

Maybe one day, haha.

File: xbmc/ServiceBroker.cpp (hunk 1 above)
Comment (fuzzard) [reply xbmc/ServiceBroker.cpp]: Not sure we should. Ideally the state of nullptr should be handled (and possibly logged) where called from.
If others want it, i can put in.

**Review by garbear — APPROVED** [review]

**Review by lrusak — APPROVED** [review]

**Comment by lrusak** [comment]
Make sure to squash and it should be g2g

**Pull request merged**